### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by zdmytriv**
## what
This GitHub Action adopts a streamlined approach to managing release branches, drawing on a trunk-based branching strategy. In this model, the `DEFAULT_BRANCH` consistently represents the most recent release, while release branches are exclusively created for previous major releases, if applicable. This structure simplifies the process for contributors when submitting Pull Requests for bug fixes or backporting modifications to older releases, as it enables them to target a specific major release.

**How it works:** upon publishing a new major release `N`, a corresponding branch for the previous release `N-1` will be automatically generated.

Imagine you have tags like this in your repo:

```
0.1.0
0.2.0
1.0.0
1.1.0
1.2.0
1.2.1
2.0.0
2.1.0
2.2.0
3.0.0
3.1.0   main
```

Upon the first release published event, the "release branch manager" will generate new branches named `release/vN-1`, where N corresponds to the latest tag of each major release. In this case, several new branches will be created:

```
0.1.0
0.2.0   release/v0
1.0.0
1.1.0
1.2.0
1.2.1   release/v1
2.0.0
2.1.0
2.2.0   release/v2
3.0.0
3.1.0   main
```

Note that `3.1.0` is latest tag and release branch manager wouldn't create release branch because latest major release is maintained in `main` branch.

If you wish to make changes to `2.2.0`, you must create a pull request for the `release/v2` branch and generate a corresponding release/tag with a major version of `2`, for example, `2.3.0`.

This action requires GitHub releases to follow the [SemVer versioning](https://semver.org/) scheme.

## why
* Initial implementation of Release Branch Manager GitHub Action

### CODE REVIEW

**Comment by zdmytriv** [comment]
@goruha as i understand it creates a release branch for the current release but what we want is to create a release branch for the last tag of the previous major tag.

for example we have git log like this
```
2023-04-14 e6de607 Test User update (HEAD -> main)
2023-04-14 e736430 Test User update
2023-04-14 142e5a5 Test User update
2023-04-14 12f8920 Test User update
2023-04-14 9d27af8 Test User update (tag: 1.1.0)
2023-04-14 5c6c40f Test User update (tag: 1.0.0)
2023-04-14 bb8e032 Test User update (tag: 0.3.5, release/v0)
...
```

now we want publish new major release `v2.0.0`. Once we do that we will get tag `2.0.0` created for last commit `e6de607`. So our git log will look like this:

```
2023-04-14 e6de607 Test User update (HEAD -> main, tag: 2.0.0)
2023-04-14 e736430 Test User update
2023-04-14 142e5a5 Test User update
2023-04-14 12f8920 Test User update
2023-04-14 9d27af8 Test User update (tag: 1.1.0)
2023-04-14 5c6c40f Test User update (tag: 1.0.0)
2023-04-14 bb8e032 Test User update (tag: 0.3.5, release/v0)
...
```
^ note tag `2.0.0` showed up

then our release branch manager workflow will kick in and it will create `release/v1` branch in correct tag `1.1.0`. So our git log will look like

```
2023-04-14 e6de607 Test User update (HEAD -> main, tag: 2.0.0)
2023-04-14 e736430 Test User update
2023-04-14 142e5a5 Test User update
2023-04-14 12f8920 Test User update
2023-04-14 9d27af8 Test User update (tag: 1.1.0, release/v1)
2023-04-14 5c6c40f Test User update (tag: 1.0.0)
2023-04-14 bb8e032 Test User update (tag: 0.3.5, release/v0)
...
```
^ note where `release/v1` points to

File: README.yaml [hunk 1]
```diff
@@ -1,78 +1,68 @@
...
+description: GitHub Action for Managing Release Branches
 
 introduction: |-
-  This is template repository to create composite GitHub Actions. 
-  Feel free to use it as reference and starting point.
+  On publishing new major release `N` branch for previous release `N-1` will be created.
+
+  For example when new release `v5.0.0` with tag `5.0.0` gets published this workflow will create new branch `release/v4`.
+
+  This action requires repository to use SemVer versioning.
```
Comment (osterman) [inline README.yaml:28]:
```suggestion
  This action requires GitHub releases to follow the [SemVer versioning](https://semver.org/) scheme.
```

File: README.yaml [hunk 2]
```diff
@@ -1,78 +1,68 @@
...
-  - name: "github-actions-workflows"
-    description: "Reusable workflows for different types of projects"
-    url: "https://github.com/cloudposse/github-actions-workflows"
-  - name: "example-github-action-release-workflow"
-    description: "Example application with complicated release workflow"
-    url: "https://github.com/cloudposse/example-github-action-release-workflow"
-
-# How to use this project
 usage: |-
+  Example of workflow that can be added to repository.
```
Comment (osterman) [inline README.yaml:31]:
```suggestion
  Example of workflow that that will create major release tags. To use it, just add this workflow to your `.github/workflows` directory. 
```

File: README.yaml [hunk 3]
```diff
@@ -1,78 +1,68 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   ```
 
+references:
+  - name: "Release Drafter GitHub Action"
+    description: 'Drafts your next release notes as pull requests are merged into your default branch.'
+    url: "https://github.com/release-drafter/release-drafter"
+  - name: "Release Branch Manager GitHub Action"
+    description: 'Automatically creates "Long Term Support (LTS)" release branches when new releases are published'
+    url: "https://en.wikipedia.org/wiki/Test_harness"
```
Comment (osterman) [inline README.yaml:59]:
```suggestion
    url: "https://github.com/cloudposse/github-action-release-branch-manager"
```

Comment (osterman) [reply README.yaml:59]: Add major release tagger

File: action.yml [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+
+    - name: Install Node Dependencies
       shell: bash
       run: |
-          echo "::set-output name=action-result::${{ inputs.param1 }}"
+        cd ${{ github.action_path }}
+        npm install
 
+    - name: Run Release Branch Manager
+      shell: bash
```
Comment (osterman) [inline action.yml:36]: Replace with github script that `includes` the index

File: action.yml [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+      id: run_release_branch_manager
+      run: |
+        cd ${{ github.action_path }}
+        export LOG_LEVEL='${{ inputs.log-level }}'
+        node index.js $GITHUB_WORKSPACE
+
+    - name: Get the output
```
Comment (osterman) [inline action.yml:43]:
```suggestion
    - name: Debug output
```

**Review by osterman — APPROVED** [review]

**Pull request merged**